weblate/settings_openshift.py

```diff
@@ -161,6 +170,26 @@ def get_env_list(name, default=None):
     AUTHENTICATION_BACKENDS += ('social_core.backends.gitlab.GitLabOAuth2',)
     SOCIAL_AUTH_GITLAB_SCOPE = ['api']
 
+if 'WEBLATE_SOCIAL_AUTH_GITHUB_KEY' in os.environ:
+    AUTHENTICATION_BACKENDS += ('social_core.backends.github.GithubOAuth2',)
+    SOCIAL_AUTH_GITHUB_SCOPE = ['user:email']
+
+if 'WEBLATE_SOCIAL_AUTH_AUTH0_KEY' in os.environ:
+    AUTHENTICATION_BACKENDS += ('social_core.backends.auth0.Auth0OAuth2',)
+    SOCIAL_AUTH_AUTH0_SCOPE = ['openid', 'profile', 'email']
+
+if 'WEBLATE_SOCIAL_AUTH_AUTH0_AUTH_EXTRA_ARGUMENTS' in os.environ:
+    auth0_args = parse_qs(ast.literal_eval(
+      Template(
+        os.environ.get('WEBLATE_SOCIAL_AUTH_AUTH0_AUTH_EXTRA_ARGUMENTS', '""')
+      ).substitute(os.environ)
+    ))
+    auth0_args = dict(
+      [(i, auth0_args[i][0]) if auth0_args[i] else (i, '') for i in auth0_args]
+    )
+    SOCIAL_AUTH_AUTH0_AUTH_EXTRA_ARGUMENTS = auth0_args
```

Comment:
I've just moved helper functions we use in Docker to Weblate, so you might better use that as well instead of using ast (though I'm not sure what kind of args this expects, `get_env_map` deals with strings only):

https://github.com/WeblateOrg/weblate/blob/e002f976000ecdea1d8e3893520aaef12d54a56e/weblate/utils/environment.py#L32-L41

Reply:
Well, it expects querystring style of params, so sth like `arg1=value1&arg2=something%20else`.

But if you decided to do it in format `arg1:value,arg2:something else` - its also fine by me, will refactor it. It makes sense it is same all around weblate.

Reply:
@nijel actuall all strings are required to be in `"some_string"`, while this one does not expect this - aka I need to set variable as `some_string`, so without **""**. 

It just doesn't seems to be consistent with `import_env_vars` which is used in `openshhiftlib.py`. Maybe also refactor that one, so all strings does not need to be wrapped in **""**?